quarkus/config-api/src/main/java/org/keycloak/config/EventOptions.java

```diff
@@ -0,0 +1,30 @@
+package org.keycloak.config;
+
+import java.util.List;
+
+public class EventOptions {
+
+    public static final Option<Boolean> USER_EVENT_METRICS_ENABLED = new OptionBuilder<>("user-event-metrics-enabled", Boolean.class)
+            .category(OptionCategory.EVENTS)
+            .description("Create metrics based on user events.")
+            .buildTime(true)
+            .defaultValue(Boolean.FALSE)
+            .build();
+
+    public static final Option<List<String>> USER_EVENT_METRICS_TAGS = OptionBuilder.listOptionBuilder("user-event-metrics-tags", String.class)
+            .category(OptionCategory.EVENTS)
+            .description("Comma-separated list of tags to be collected for event metrics. By default only 'realm' is enabled to avoid a high metrics cardinality.")
+            .buildTime(false)
+            .expectedValues(List.of("realm", "idp", "clientId"))
+            .defaultValue(List.of("realm"))
+            .build();
+
+    public static final Option<List<String>> USER_EVENT_METRICS_EVENTS = OptionBuilder.listOptionBuilder("user-event-metrics-events", String.class)
+            .category(OptionCategory.EVENTS)
+            .description("Comma-separated list of events to be collected for event metrics. Reduce the number of metrics. If empty or not set, all events create a metric.")
+            .buildTime(false)
+            .build();
```

Comment:
I chose the name `user-event-metrics-enabled` to with the names of other metrics related `http-metrics-slos`,  `http-metrics-histograms-enabled` and `cache-metrics-histograms-enabled`. 

Which events and which tags to add is something that is IMHO then specific to events, not not necessarily to metrics. 

Eventually we would have additional add admin events as well (if we agree on it), and something that I hope for is that we would allow storing events in an external event store like InfluxDB, and those would go into the events category. 

So that's why I chose the "Event" category. 

If we move the options to the "Metrics" category, I would at least keep their names so we can then later move it to the "Event" category. 

Looking forward to hear your thoughts.

Reply:
@ahus1 Thank you for elaborating on reasoning behind choosing the names for these new options.

My feeling was it felt a bit random to introduce a new category and a `user-event-*` prefix just for new metrics. But I understand your point that there's a strong precedence in other metrics options that are placed in other categories and prefixes like http or cache.

Looking at it in retrospective, I feel this might not have been an optimal solution. Despite it felt natural at the moment to place a metrics related option to a category the specific metric is getting data from, now with the increasing number of various metrics toggles it might feel confusing and overwhelming to user. As a user I'd probably expect to see at a single place (a single category/prefix) what metrics I can toggle and what config options do I have for tweaking tags etc.

If we want to keep the current naming paradigm (which I agree is probably our only option now if we don't want to do the dance around deprecating/renaming existing metrics related options), I'd suggest to make the events category more generic:
* Rename the config category from "User Event Metrics" to just "Events", and perhaps also adjust its description to be generic for Events and not just Metrics.
* Change the prefix of the new options to `events-metrics-user-*`. That would align it with existing options like `http-metrics-*` and `cache-metrics-*`.

Let me know your thoughts.

Reply:
Updated in next commit

Reply:
Removed one `s` as it seemed not to be necessary. Example: `event-metrics-user-enabled` - See below for all option: 

```
Events (Preview):

--event-metrics-user-enabled <true|false>
                     Preview: Create metrics based on user events. Default: false. Available only
                       when metrics are enabled and feature user-event-metrics is enabled.
--event-metrics-user-events <events>
                     Preview: Comma-separated list of events to be collected for user event
                       metrics. Reduce the number of metrics. If empty or not set, all events
                       create a metric. Available only when user event metrics are enabled.
--event-metrics-user-tags <tags>
                     Preview: Comma-separated list of tags to be collected for user event metrics.
                       By default only 'realm' is enabled to avoid a high metrics cardinality.
                       Possible values are: realm, idp, clientId. Default: realm. Available only
                       when user event metrics are enabled.
```

Please let me know if this matches your expectations